src/testHelpers.ts

```diff
@@ -40,11 +40,16 @@ export function initTestHelpers() {
 
   function setup() {
     if (isJSDOMEnvironment()) {
-      // Remove initial JSDOM <head> element
-      const headElement = document.querySelector('head');
-      if (headElement) {
-        headElement.remove();
-      }
+      // This is needed by NextJS to correctly find the head element which is rendered in `document.body`
+      // https://github.com/vercel/next.js/blob/c8cd77a856248346768042f4553a8513195311a4/packages/next/client/head-manager.ts#L36
+      const getElementsByTagName = document.getElementsByTagName;
```

Comment:
I'm not fully into the context of this, but I expect a lot of troubles coming from patching global methods.

Could it be an alert the we're doing something wrong? (Let's forget for a moment that rendering a `Next.js` app in JSDOM is already a wrong idea per se 😄 ).

Reply:
I think we are doing best we can with what we have (JSDOM) 😄 Not sure if you saw my comment why this was needed: https://github.com/toomuchdesign/next-page-tester/pull/76/files#r550180943

Reply:
Now I got full context about this part. The trade-off is: always render an initial document with missing `head` vs. patching global `document.getElementsByTagName` to tweak the way Next.js retrieve pages' `head` element. 🤔